core/src/main/java/org/jdbi/v3/core/statement/ParsedParameters.java

```diff
@@ -78,4 +79,35 @@ public String toString() {
             + ", parameterNames=" + parameterNames
             + '}';
     }
+
+    /**
+     * A static factory of named {@link ParsedParameters} instances.
+     * List given in argument must contain parameter names from SQL.
+     * Names from a list must be stripped of a colon which stands
+     * before variable in unprocessed SQL.
+     *
+     * @param names the parameter names from SQL statement
+     * @return New {@link ParsedParameters} instance
+     * @throws IllegalArgumentException if names list contains positional parameter
+     */
+    public static ParsedParameters named(List<String> names) {
+        if (names.contains(ParsedSql.POSITIONAL_PARAM)) {
+            throw new IllegalArgumentException("Named parameters "
```

Comment:
I think `UnableToCreateStatementException` would be better here

Reply:
Why is that? It's not statement being created here.

Reply:
Well sort of. The exceptions indicate in roughly which part of the overall jdbi usage process something has gone wrong. If you disagree, I'm fine with passing this up, I think we're not really icky about changing exceptions at a later time.

Reply:
What you stated is valid, however I do not think that changing exception class in this factory is a solution. From what I see in the code the process is started in `SqlStatement.internalExecute()` and involves the following steps: rendering, parsing, statement creation and customization, but what is interesting is the fact that only statement creation and customization is guarded by `try/catch` block which catch `SQLException` and wraps it with `UnableToCreateStatementException` (because it had to be caught).

If we consider the fact that custom implementations of `TemplateEngine`, `SqlParser`, `StatementCustomizer`, etc, are not enforced to throw `UnableToCreateStatementException`, then, with above in mind, and in regards to your comment, we can conclude that any custom implementation which throws non-statement creation exception, breaks indication you described. With existing impl this indication is only assumed - it works only when default components are used. Using custom components does not guarantee this indication at all.

I believe this could be fixed by doing general exception handling in `SqlStatement`.`internalExecute()` and wrapping exception thrown by custom implementations the same way `SQLException` is done. This would give us real, customization-resistant indication of where and why process failed.

I would like to leave this `IllegalArgumentException` as it is, but I'm completely ok if JDBI team disagree and decide change this later. I'm not really attached to my code :)

Reply:
Well yeah, but just because we can't control every external factor doesn't mean we should neglect to control our internal factors. But let's just leave it as it is now, it's not a big deal and I don't want to keep nitpicking on you over things that are of little import.

Reply:
Ok, marking this one as resolved.